ci/cscs-ci.yml

```diff
@@ -1,185 +1,139 @@
+#
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2024, ETH Zurich
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
 include:
   - remote: 'https://gitlab.com/cscs-ci/recipes/-/raw/master/templates/v2/.ci-ext.yml'
+  - local: 'ci/cscs-ci-ext-config.yml'
 
-.py313: &py313
-  PYVERSION_PREFIX: py313
-  PYVERSION: 3.13.3
-.py310: &py310
-  PYVERSION_PREFIX: py310
-  PYVERSION: 3.10.9
+variables:  # Default values for base variables (can be overriden in jobs definitions)
+  CUDA_VERSION: '12.6.2'
+  ROCM_VERSION: '6.2.4'
+  UBUNTU_VERSION: '24.04'
+  UV_VERSION: '0.6.12'
 
+.test_python_versions: &test_python_versions ['3.10', '3.13']
 
 stages:
-  - baseimage
-  - image
+  - build
   - test
 
-.build_baseimage:
-  stage: baseimage
-  # we create a tag that depends on the SHA value of ci/base.Dockerfile, this way
-  # a new base image is only built when the SHA of this file changes
-  # If there are more dependency files that should change the tag-name of the base container
-  # image, they can be added too.
-  # Since the base image name is runtime dependent, we need to carry the value of it to
-  # the following jobs via a dotenv file.
+# -- Build stage --
+# The build job creates a the base image for the CSCS CI only if the external
+# dependencies change. The actual repository is cloned at the test stage.
+.build_common:
+  stage: build
+  variables:
+    BASE_IMAGE: docker.io/ubuntu:${UBUNTU_VERSION}
```

Comment:
This could be `jfrog.svc.cscs.ch/dockerhub/ubuntu:${UBUNTU_VERSION}`, which is proxied through JFrog, and cached. Not really a big saving when it comes to network, when the image is as tiny as an ubuntu base image.